src/patchelf.cc

```diff
@@ -1421,7 +1442,7 @@ static void patchElf2(ElfFile & elfFile, mode_t fileMode)
 
 static void patchElf()
 {
-    if (!printInterpreter && !printRPath && !printSoname)
+    if (!printInterpreter && !printRPath && !printSoname && !printNeeded)
```

Comment:
This seem incorrect. Printing the needed section will not avoid manupilating the file.